### PR TITLE
WIP: Add JSON-RPC interface documentation

### PR DESCRIPTION
**Pull request by promag**
This is poorly written and structured but I want to know if it makes sense.

The idea is not to copy `-help` outputs.

Closes #14448.

Follow ups:
 - explain named parameters
 - detail how to select the wallet, how -rpcwallet works, etc*
 - explain how deprecation works
 - https://github.com/bitcoin/bitcoin/pull/14458#issuecomment-456202424 
 - https://github.com/bitcoin/bitcoin/pull/14458#discussion_r257926917
 - add related configurations while taking into account https://github.com/bitcoin/bitcoin/pull/14458#issuecomment-454432287

### CODE REVIEW

**Comment by kristapsk** [comment]
Concept ACK

**Comment by ch4ot1c** [comment]
Concept ACK 👍

**Comment by sipa** [comment]
Concept ACK

File: doc/JSON-RPC-interface.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+JSON-RPC Interface
+==================
+
+The JSON-RPC API is by default enabled in bitcoind and disabled on bitcoin-qt.
+This can be changed with `-noserver` and `-server` options. In bitcoin-qt it is
+possible to issue execute RPC methods in the Debug Console Dialog.
```
Comment (jnewbery) [inline doc/JSON-RPC-interface.md]: Remove 'issue '

**Review by jnewbery — COMMENTED** [review]
Concept ACK. Looks good so far!

File: doc/JSON-RPC-interface.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
+JSON-RPC Interface
```
Comment (practicalswift) [inline doc/JSON-RPC-interface.md]: Please rename this file `json-rpc-interface.md` to make it consistent with the other files.

File: doc/JSON-RPC-interface.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+The JSON-RPC API is by default enabled in bitcoind and disabled on bitcoin-qt.
+This can be changed with `-noserver` and `-server` options. In bitcoin-qt it is
+possible to issue execute RPC methods in the Debug Console Dialog.
+
+*Add some basic examples like:*
+
+*To see the available methods use `bitcoin-cli -regtest help`*
+
+#### Server Configuration
```
Comment (practicalswift) [inline doc/JSON-RPC-interface.md]: Increment header levels one level at a time. You might want to use `mdl` to check the Markdown formatting :-)

File: doc/JSON-RPC-interface.md (hunk 2 above)
Comment (ch4ot1c) [reply doc/JSON-RPC-interface.md]: I think matching `REST-interface.md` is the intention - should it be lowercase too?

**Comment by laanwj** [comment]
@harding might want to have a look here as this adds technical documentation

**Comment by promag** [comment]
Rebased, still WIP though.

File: doc/JSON-RPC-interface.md [hunk 4]
```diff
@@ -2,8 +2,8 @@
 
 The headless daemon `bitcoind` has the JSON-RPC API enabled by default, the GUI
 `bitcoin-qt` has it disabled by default. This can be changed with the `-server`
-option. In the GUI it is possible to execute RPC methods in the Debug Console
-Dialog.
+or `-noserver` options. In the GUI it is possible to execute RPC methods in the
```
Comment (MarcoFalke) [inline doc/JSON-RPC-interface.md]: noserver is not a separate option

File: doc/JSON-RPC-interface.md [hunk 5]
```diff
@@ -36,3 +36,113 @@ However, the wallet may not be up-to-date with the current state of the mempool
 or the state of the mempool by an RPC that returned before this RPC. For
 example, a wallet transaction that was BIP-125-replaced in the mempool prior to
 this RPC may not yet be reflected as such in this RPC response.
+
+## Basic usage
+
+Launch the daemin in the background:
```
Comment (fanquake) [inline doc/JSON-RPC-interface.md]: s/daemin/daemon

Comment (Sjors) [inline doc/JSON-RPC-interface.md]: Nit / later: explain that as an alternative you can run the daemon in one tab and use bitcoin-cli in another. That way you can keep an eye on logs. Or just mention `tail -f <datadir>/<testnet3 or regtest>/debug.log`

See also #15438.

File: doc/JSON-RPC-interface.md [hunk 6]
```diff
@@ -36,3 +36,113 @@ However, the wallet may not be up-to-date with the current state of the mempool
...
+
+-rpcwait
+     Wait for RPC server to start
+
+-rpcwallet=<walletname>
+     Send RPC for non-default wallet on RPC server (needs to exactly match
+     corresponding -wallet option passed to bitcoind)
+```
+
+*Also talk about named parameters?*
```
Comment (Sjors) [inline doc/JSON-RPC-interface.md]: This should go away if we merge it for for the 0.18 release.

**Comment by Sjors** [comment]
ACK. I suggest we remove the "todo" comment, merge this and improve it after the 0.18 release.

**Review by Sjors — COMMENTED** [review]
ACK. I suggest we remove the "todo" comments, merge this and improve it after the 0.18 release.

**Comment by promag** [comment]
Rebased and removed configurations section, kind of agree with @Sjors https://github.com/bitcoin/bitcoin/pull/14458#issuecomment-454432287. What left isn't worth I think, so closing.

**Pull request closed without merging**